aiidalab/app.py

```diff
@@ -855,36 +855,21 @@ def _is_compatible(self, app_version: str) -> bool:
         except KeyError:
             return False  # compatibility indetermined for given version
 
-    def _remote_update_status(self) -> bool:
-        """Determine whether there are updates available.
-
-        For this the app must be installed in a known version and there must be
-        available (and compatible) versions.
-        """
-        installed_version = self._installed_version()
-        if installed_version not in (AppVersion.UNKNOWN, AppVersion.NOT_INSTALLED):
-            available_versions = list(self.available_versions)
-            if len(available_versions):
-                return self._installed_version() != available_versions[0]  # type: ignore
-        return False
-
     def _refresh_versions(self) -> None:
         self.installed_version = self._installed_version()
         self.include_prereleases = self.include_prereleases or (
             isinstance(self.installed_version, str)
             and parse(self.installed_version).is_prerelease
         )
 
-        all_available_versions = list(self._app.available_versions())
+        all_available_versions = list(self._app.available_versions(prereleases=True))
         self.has_prereleases = any(
             parse(version).is_prerelease for version in all_available_versions
         )
-        if self._app.is_registered:
```

Comment:
What do you mean? This is not changed but moved to the `self._app.available_versions`, you can find the `is_registered` still there.

Reply:
I see. But have you tested what will happen? It seems like `available_versions` will return None in that case, and you then apply `list` on that which would throw an exception. Please add a test for this.

Reply:
Yes, it will give empty generator and still valid to be used. please see the test I implemented at https://github.com/aiidalab/aiidalab/pull/368/commits/dc9bf4972b00e09b8ac044b2e686a5b2be913ef1

Reply:
Something not correct, the test passed locally but not remotely.

Reply:
The problem comes from I have the "quantum-espresso" app installed locally and the mocked object is regarded as installed since the app path is existed even I set the `is_registred` False. 
The issue is rather subtle that in the `installed_version` method, it uses `path.exist()` rather the `is_installed`, I monckpatch the `is_installed` in the test but the `installed_version` does not notice the patch.
It works as expected now.

Reply:
Changed.